proposals/specification/access-consents.bs

```diff
@@ -0,0 +1,820 @@
+      [=Access Needs=] that the [=Access Consent=] is
+      satisfying</td>
+    </tr>
+    <tr>
+      <td>hasDataConsent</td>
+      <td>[DataConsent](#classDataConsent)</td>
+      <td>Consent for a specific type of data</td>
+    </tr>
+  </tbody>
+</table>
+
+<figure>
+  <figcaption>Alice's [=Access Consent=] for Projectron at  
+    https://alice.example/consents/e2765d6c -
+    <a href="snippets/alice.example/e2765d6c.ttl">View</a>
+  </figcaption>
+  <pre class=include-code>
+  path: snippets/alice.example/e2765d6c.ttl
+  highlight: turtle
+  show: 11-21
+  </pre>
+</figure>
+
+<figure>
+  <figcaption>Alice's [=Access Consent=] for Bob at  
+    https://alice.example/consents/4460dce3 -
+    <a href="snippets/alice.example/4460dce3.ttl">View</a>
+    </figcaption>
+  <pre class=include-code>
+  path: snippets/alice.example/4460dce3.ttl
+  highlight: turtle
+  show: 11-21
+  </pre>
+</figure>
+
+## Data Consent ## {#data-consent}
+
+A <dfn>Data Consent</dfn> records the decision of a [=Social Agent=]
+to grant access to a specific type of data in their control, identified
+by a [=Shape Tree=]. They are always associated with a single
+[=Access Consent=].
+
+[=Data Consents=] should not be shared with the [=Agent=] that has 
+been granted access. [=Data Grants=] are generated from [=Data Consents=],
+and are appropriate to share with the grantee.
+
+[Class Definition](interop.ttl#DataConsent) -
+[Shape Definition](interop.shex#DataConsentShape) -
+[Shape Tree Definition](interop.tree#DataConsentTree)
+
+<table class="classinfo data" align="left" id="classDataConsent">
+  <colgroup></colgroup>
+  <colgroup></colgroup>
+  <colgroup></colgroup>
+  <thead>
+    <tr>
+      <th>Property</th>
+      <th>Range</th>
+      <th>Description</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td>dataOwner</td>
+      <td>[SocialAgent](#classSocialAgent)</td>
+      <td>[=Social Agent=] that owns the data being consented</td>
+    </tr>
+    <tr>
+      <td>registeredShapeTree</td>
+      <td><a href="https://www.w3.org/ns/shapetrees#ShapeTree" class="vocab">st:ShapeTree</a></td>
+      <td>[=Shape Tree=] representing the type of data being consented</td>
+    </tr>
+    <tr>
+      <td>accessMode</td>
+      <td>acl:Read, acl:Write, acl:Control, acl:Append</td>
+      <td>Modes of access granted to the consented data</td>
+    </tr>
+    <tr>
+      <td>creatorAccessMode</td>
+      <td>`acl:Read, acl:Write, acl:Control, acl:Append`</td>
+      <td>Additional access mode assigned to the creator of a
+        data instance. Adds to the set of modes linked via 
+        `interop:accessMode`. Only valid when `accessMode` includes
+         `acl:Write` or `acl:Append`</td>
+    </tr>
+    <tr>
+      <td>scopeOfConsent</td>
+      <td> interop:AllInstances, interop:SelectedInstances,
+      interop:InheritInstances, interop:NoAccess</td>
+      <td>Identifies the access scope of the [=Data Consent=]</td>
+    </tr>
```

Comment:
I think we should add here `hasDataRegistration` and once we clarify scopes we will create one shape per scope similar to #109 which will clarify which properties are required for which scope.

Reply:
Resolved in https://github.com/solid/data-interoperability-panel/commit/4447bd7d28b79e7100ff2af76eda2677ca48fb29